llama.cpp

```diff
@@ -3729,19 +3729,16 @@ static bool llm_load_tensors(
         model.tensors_by_name.emplace_back(ggml_get_name(cur), cur);
     }
 
-    ok = ok && ml.load_all_data(ctx, progress_callback, progress_callback_user_data, buf_mmap, use_mlock ? &model.mlock_mmap : NULL);
-    if (progress_callback) {
-        // Even though the model is done loading, we still honor
-        // cancellation since we need to free allocations.
-        ok = ok && progress_callback(1.0f, progress_callback_user_data);
+    if (!ml.load_all_data(ctx, progress_callback, progress_callback_user_data, buf_mmap, use_mlock ? &model.mlock_mmap : NULL)) {
+        return false;
     }
 
     model.mapping = std::move(ml.mapping);
```

Comment:
@slaren do you know if this line will be a problem? Since it doesn't get run if the above returns early

Reply:
> @slaren do you know if this line will be a problem? Since it doesn't get run if the above returns early

The progress callback should only be called if it loaded successfully, I think. Would be weird to run it with 1.0 if the model load actually failed

Reply:
Skipping the `mapping` move should be fine.

Reply:
The case I'm trying to **avoid** is:
1. The model has fully loaded
2. The client has received some information that the model load should be cancelled (e.g. on another thread)
3. llama.cpp calls the progress_callback with `1.0f`
4. Client returns false (independent from the `1.0f` value itself)
5. llama.cpp ignores the callback return value, and returns a valid `llama_model *`
6. Now the client has to free the model, even though it did not expect one to be returned